lib/api/controllers/authController.js

```diff
@@ -76,20 +74,16 @@ class AuthController {
    * @returns {Promise<Token>}
    */
   login(request) {
-    assertHasBody(request);
-
-    if (!request.input.args.strategy) {
-      throw new BadRequestError('No strategy specified');
-    }
+    assertArgsHasAttribute(request, 'strategy');
 
     const strategy = request.input.args.strategy;
 
-    return this.kuzzle.passport.authenticate({query: request.input.body, original: request}, strategy)
+    return this.kuzzle.passport.authenticate({body: request.input.body, query: request.input.args}, strategy)
```

Comment:
`passport.authenticate` want as first argument a JSON object representing an HTTP request, as described in Express: http://expressjs.com/fr/api.html#req
Then, it's the responsibility of each strategy to use `query` or `body` attributes, depending of its needs.